cachers/combined.go

```diff
@@ -0,0 +1,161 @@
+package cachers
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+)
+
+// CombinedCache is a LocalCache that wraps a LocalCache and a RemoteCache.
+// It also keeps times for the remote cache Download/Uploads
+type CombinedCache struct {
+	verbose     bool
+	localCache  LocalCache
+	remoteCache RemoteCache
+	putsMetrics *timeKeeper
+	getsMetrics *timeKeeper
+}
+
+func (l *CombinedCache) Kind() string {
+	return "combined"
+}
+
+func NewCombinedCache(localCache LocalCache, remoteCache RemoteCache, verbose bool) LocalCache {
+	cache := &CombinedCache{
+		verbose:     verbose,
+		localCache:  localCache,
+		remoteCache: remoteCache,
+		putsMetrics: newTimeKeeper(),
+		getsMetrics: newTimeKeeper(),
+	}
+	if verbose {
+		cache.localCache = NewLocalCacheStates(localCache)
+		cache.remoteCache = NewRemoteCacheStats(remoteCache)
+		return NewLocalCacheStates(cache)
+	}
+	return cache
+}
+
+func (l *CombinedCache) Start() error {
+	err := l.localCache.Start()
+	if err != nil {
+		return fmt.Errorf("local cache start failed: %w", err)
+	}
+	err = l.remoteCache.Start()
+	if err != nil {
+		return fmt.Errorf("remote cache start failed: %w", err)
+	}
+	l.putsMetrics.Start()
+	l.getsMetrics.Start()
+	return nil
+}
+
+func (l *CombinedCache) Close() error {
+	err := l.localCache.Close()
+	if err != nil {
+		err = fmt.Errorf("local cache stop failed: %w", err)
+	}
+	err = l.remoteCache.Close()
+	if err != nil {
+		err = errors.Join(fmt.Errorf("remote cache stop failed: %w", err))
+	}
+	l.putsMetrics.Stop()
+	l.getsMetrics.Stop()
+	if l.verbose {
+		log.Printf("[%s]\tDownloads: %s, Uploads %s", l.remoteCache.Kind(), l.getsMetrics.Summary(), l.putsMetrics.Summary())
+	}
+	return err
+}
+
+func (l *CombinedCache) Get(ctx context.Context, actionID string) (outputID, diskPath string, err error) {
```

Comment:
can probably avoid the named parameters in output, IMO it's confusing
```suggestion
func (l *CombinedCache) Get(ctx context.Context, actionID string) (string, string, error) {
```